integration/common/openlineage/common/provider/bigquery.py

```diff
@@ -87,20 +87,23 @@ def __init__(
         client: Optional[Client] = None,
         logger: Optional[logging.Logger] = None
     ):
-        self.client = client
         if client is None:
             self.client = Client()
-        self.logger = logger
+        else:
+            self.client = client
         if logger is None:
-            self.logger = logging.getLogger(__name__)
+            self.logger: logging.Logger = logging.getLogger(__name__)
+        else:
+            self.logger = logger
 
     def get_facets(self, job_id: str) -> BigQueryFacets:
         inputs = []
         output = None
         run_facets = {}
         try:
             try:
-                job = self.client.get_job(job_id=job_id)
+                if self.client:
```

Comment:
After `__init__`, the `self.client` will never be None. I don't understand why it complains about this :) 

Maybe it's better to disable check here instead of confusing people who will look where the client can be unset.